packages/core/src/internal/fetchSource.ts

```diff
@@ -57,21 +24,44 @@ export const makeFetchSource = (
           fetchOptions.signal = abortController.signal;
         }
 
-        return executeFetch(operation, url, fetchOptions);
+        // Does the fetch call — handling any network level errors, like not 200's
+        return fetcher(url, fetchOptions).then((response: Response) => {
+          catch_response = response;
+
+          if (
+            response.status < 200 ||
+            response.status >= (fetchOptions.redirect === 'manual' ? 400 : 300)
+          ) {
+            throw new Error(response.statusText);
+          }
+
+          return response;
+        });
       })
-      .then((result: OperationResult | undefined) => {
-        if (!ended) {
-          ended = true;
-          if (result) next(result);
-          complete();
-        }
-      });
+      // now we handle the maybe response
+      .then(async (response: Response | undefined) => {
+        if (!response || ended) return;
+
+        const networkResult = await response.json();
```

Comment:
@kitten  this is the line where we'd do the `meros` checks with the mime. (which is [built into meros](https://github.com/maraisr/meros/blob/main/src/browser.ts#L27) anyway)